driver.go

```diff
@@ -73,7 +72,7 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 		mc.netConn, err = nd.Dial(mc.cfg.Net, mc.cfg.Addr)
```

Comment:
I think we should use `nd.DialContext()` when context is not nil.

Reply:
The connector change would not even travel down this path, and this change seems like it belongs in its own enhancement. When calling the connector it will use sql.OpenDB which then travels down thru the Connect function. I mean if we want to add a context here and thread that down into the connectServer then I think we could do that and have context support in both the Open and the OpenDB however I am not really sure what that gains us.

Reply:
I will look at it. I mis-understood this diff on my previous comment. Seems reasonable enough to thread the context thru however I am not sure the best way to do that off hand as the other entry point sql.Open does carry any context and uses this same connectServer function.